browser/src/react-redux/App/Signup/SignupForm.jsx

```diff
@@ -0,0 +1,156 @@
+import React, { Component } from "react";
+import axios from "axios";
+import Form from "./Form";
+import utility from "./signupUtility";
+
+class signupForm extends Component {
+  constructor() {
+    super();
+    this.state = {
+      email: "",
+      password: "",
+      confirmPassword: "",
+      errors: {
+        confirmPassword: "",
+        password: "",
+        email: "",
+        general: ""
+      },
+      buttonDisabled: false
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  handleChange(key, e) {
+    this.setState({
+      [key]: e
+    });
+  }
+  onSubmit(e) {
+    this.setState({
+      errors: {
+        general: "",
+        email: "",
+        password: ""
+      }
+    });
+
+    e.preventDefault();
+
+    const user = {
+      email: this.state.email,
+      password: this.state.password
+    };
+
+    axios
```

Comment:
To keep it at 1 level of nesting, let's go ahead and move the post up, like this:

```
axios.post("/api/user", user)
```

Otherwise, because post returns a promise, it would make sense to have 2 indents for `.then` and ..let's just keep it with 1 indent.